lib/core/application/wallet/wallet_bloc/wallet_bloc.dart

```diff
@@ -1,61 +1,98 @@
-import 'package:app/core/service/wallet_connect/wallet_connect_service.dart';
+import 'package:app/core/domain/wallet/wallet_repository.dart';
+import 'package:app/core/service/wallet/wallet_connect_service.dart';
 import 'package:app/core/utils/snackbar_utils.dart';
+import 'package:app/core/utils/web3_utils.dart';
+import 'package:app/injection/register_module.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
 import 'package:walletconnect_flutter_v2/apis/sign_api/models/session_models.dart';
 
 part 'wallet_bloc.freezed.dart';
 
 class WalletBloc extends Bloc<WalletEvent, WalletState> {
-  final WalletService walletService = WalletService();
+  final walletRepository = getIt<WalletRepository>();
+  final walletConnectService = getIt<WalletConnectService>();
 
   WalletBloc() : super(const WalletState()) {
     on<WalletEventInitWalletConnect>(_onInitWalletConnect);
     on<WalletEventGetActiveSessions>(_onGetActiveSessions);
     on<WalletEventConnectWallet>(_onConnectWallet);
     on<WalletEventUpdateUserWallet>(_onUpdateUserWallet);
+    on<WalletEventDisconnectWallet>(_onDisconnectWallet);
   }
 
   @override
   close() async {
-    walletService.close();
+    walletConnectService.close();
     super.close();
   }
 
   _onInitWalletConnect(WalletEventInitWalletConnect event, Emitter emit) async {
-    final result = await walletService.initWallet();
-    if (result.isRight()) {
+    final success = await walletConnectService.init();
+    if (success) {
       add(const WalletEventGetActiveSessions());
     }
   }
 
   _onGetActiveSessions(WalletEventGetActiveSessions event, Emitter emit) async {
-    final activeSession = await walletService.getActiveSession();
-    if (activeSession != null) {
-      emit(state.copyWith(activeSession: activeSession));
-    }
+    final activeSession = await walletConnectService.getActiveSession();
+    emit(state.copyWith(activeSession: activeSession));
   }
 
   _onConnectWallet(WalletEventConnectWallet event, Emitter emit) async {
-    var result = await walletService.connectWallet(walletApp: event.walletApp);
-    if (result.isRight()) {
+    var success =
+        await walletConnectService.connectWallet(walletApp: event.walletApp);
+    if (success) {
       add(const WalletEvent.getActiveSessions());
     }
   }
 
+  _onDisconnectWallet(WalletEventDisconnectWallet event, Emitter emit) async {
+    await walletConnectService.disconnect();
+    add(const WalletEvent.getActiveSessions());
+  }
+
   _onUpdateUserWallet(WalletEventUpdateUserWallet event, Emitter emit) async {
-    final result = await walletService.updateUserWallet(wallet: event.wallet);
+    final walletRequestResult = await walletRepository.getUserWalletRequest(
+      wallet: event.wallet.toLowerCase(),
+    );
+
+    if (walletRequestResult.isLeft()) {
+      return SnackBarUtils.showErrorSnackbar("Sign wallet failed");
+    }
```

Comment:
We should not handle UI changes or dialog here since all of this UI need to be required on BuildContext. Prefer emit a new error state and handle it on UI side.

Reply:
Same as all `SnackBarUtils`

Reply:
👍